### DISCUSSION TITLE
AI generated code  map for navigation and context

### DISCUSSION THREAD

**Discussion by AriMKatz** [zed-industries/zed#23878] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Instead of minimap/navigating symbols, you have AI "folding" where you abstract code into a map or code graph that has a dynamic zoom feature for granularity. This is useful for the user, but also as context for the assistant as a smart "codebase summary" (see how aider does a static version)

Possibly better than fold because you aren't folding on symbols but on other possibly more abstract/ useful  things like functionality.

**Comment by g0t4** [comment]
I like the idea of this, why not let the AI access tools (to gather context) and have them be available to zoom out/in on a given codebase?!
